### PR TITLE
Refs #34385 - use `Foreman::Plugin.installed?` instead of `defined?`

### PR DESCRIPTION
**Pull request by Ron-Lavi**

### CODE REVIEW

File: lib/foreman_openscap/engine.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module ForemanOpenscap
   def self.with_katello?
-    defined?(::Katello)
+    Foreman::Plugin.installed?("Katello")
```
Comment (ezr-ondrej) [inline lib/foreman_openscap/engine.rb:3]: Is the plugin really named 'Katello'? :thinking:

Comment (evgeni) [reply lib/foreman_openscap/engine.rb:3]: I think it's `katello`

File: test/test_plugin_helper.rb [hunk 2]
```diff
@@ -5,14 +5,14 @@
 FactoryBot.definition_file_paths << File.join(File.dirname(__FILE__), 'factories')
 # Add factories from foreman_ansible
 FactoryBot.definition_file_paths << File.join(ForemanAnsible::Engine.root, '/test/factories')
-FactoryBot.definition_file_paths << File.join(ForemanPuppet::Engine.root, '/test/factories') if defined?(ForemanPuppet)
+FactoryBot.definition_file_paths << File.join(ForemanPuppet::Engine.root, '/test/factories') if Foreman::Plugin.installed?("foreman_puppet")
```
Comment (ezr-ondrej) [inline test/test_plugin_helper.rb:8]: this is quite soon in the init proces and just for tests, I'd go with `defined?(ForemanPuppet::Engine`)` here, but probably both will work, so I'm fine both ways ^_^

File: lib/foreman_openscap/engine.rb (hunk 1 above)
Comment (ezr-ondrej) [reply lib/foreman_openscap/engine.rb:3]: yup, it is: https://github.com/Katello/katello/blob/4a363fd945d56ad48f6a30446d9c16320819d4a5/lib/katello/plugin.rb#L5

Comment (Ron-Lavi) [reply lib/foreman_openscap/engine.rb:3]: just `katello` lowered case though :O

**Review by ezr-ondrej — APPROVED** [review]
LGTM :+1:

**Comment by Ron-Lavi** [comment]
@lzap @adiabramovitch could you take a look?

File: test/test_plugin_helper.rb [hunk 3]
```diff
@@ -5,14 +5,14 @@
 FactoryBot.definition_file_paths << File.join(File.dirname(__FILE__), 'factories')
 # Add factories from foreman_ansible
 FactoryBot.definition_file_paths << File.join(ForemanAnsible::Engine.root, '/test/factories')
-FactoryBot.definition_file_paths << File.join(ForemanPuppet::Engine.root, '/test/factories') if defined?(ForemanPuppet)
+FactoryBot.definition_file_paths << File.join(ForemanPuppet::Engine.root, '/test/factories') if defined?(ForemanPuppet::Engine)
```
Comment (adiabramovitch) [inline test/test_plugin_helper.rb:8]: Just out of curiosity, Why are these changes needed if you stayed with the ```defined?``` method?

Comment (Ron-Lavi) [reply test/test_plugin_helper.rb:8]: As @ezr-ondrej mentioned: https://github.com/theforeman/foreman_openscap/pull/526#discussion_r798525310

Comment (adiabramovitch) [reply test/test_plugin_helper.rb:8]: Than LGTM :)

**Comment by ares** [comment]
Thanks @Ron-Lavi and all the reviewers, merging!
